### DISCUSSION TITLE
Check Collision module problem

### DISCUSSION THREAD

**Discussion by adrimg89** [cwapi3d/cwapi3dpython#211] [Q&A]
Hi everyone.
I have detected a problem with the "check_collision" module.
I have isolated two elements that cause a collision (cadwork clash check detected it without problems), but when I run my python script, the function 'check_if_elements_are_in_collision' does not detect the collision.
Does anyone of you have the same problem?

The problem happens when there are two elements with the same geometry at the same place. It seems that if the objects are exactly at the same point it doesn't detect as a collision, but if you move one of them minimally, then it detects collision.

Thank you very much. Kind regards.

**Comment by Brunner246** [comment]
Hi @adrimg89 
Please add example code and upload a test 3d file.

**Comment by adrimg89** [comment]
Hi @Brunner246 , in attachment you will find the example files.
Thank you in advance. 
[Test.zip](https://github.com/user-attachments/files/17202048/Test.zip)

**Comment by Brunner246** [comment]
Hi @adrimg89 

Your problem is caused by corrupted data.

![image](https://github.com/user-attachments/assets/6002e8ae-196e-4709-8125-036019fad96b)

**Reply by adrimg89** [reply]
Oops, I didn't realize about it!
Thank you very much for your fast response!!
